pyccl/core.py

```diff
@@ -276,252 +296,219 @@ def __getitem__(self, key):
-                       n_s==None, Omega_k==0., Neff==3.046, m_nu==0., 
-                       mnu_type==None, w0==-1., wa==0., 
-                       bcm_log10Mc==np.log10(1.2e14), bcm_etab==0.5, 
-                       bcm_ks==55., sigma8==None, z_mg==None, df_mg==None]
-            
-            if not all(arg == True for arg in argtest):
-                warn("Cosmological parameter kwargs are ignored if 'params' is "
-                     "not None", UserWarning)
+            if kwargs:
+                warn(
+                    "Cosmological parameter kwargs are ignored if 'params' is "
+                    "not None", UserWarning)
         else:
             raise TypeError("'params' is not a valid Parameters object.")
-        
+
         # Check that the ccl_configuration-related arguments are valid
         if config is not None:
-            # User passed a ccl_configuration object; ignore other arguments 
+            # User passed a ccl_configuration object; ignore other arguments
             # and use this
-            
+
             # Check that input object is of the correct type
             if not isinstance(config, lib.configuration):
                 raise TypeError("'config' is not a valid ccl_configuration "
                                 "object.")
-            
+
             # Store ccl_configuration for later access
             self.configuration = config
-            
+
         else:
             # Construct a new ccl_configuration object from kwargs
-            
+
             # Check validity of configuration-related arguments
             if transfer_function not in transfer_function_types.keys():
-                raise KeyError( "'%s' is not a valid transfer_function type. "
-                                  "Available options are: %s" \
-                                 % (transfer_function, 
-                                    transfer_function_types.keys()) )
+                raise ValueError(
+                    "'%s' is not a valid transfer_function type. "
+                    "Available options are: %s"
+                    % (transfer_function,
+                       transfer_function_types.keys()))
             if matter_power_spectrum not in matter_power_spectrum_types.keys():
-                raise KeyError( "'%s' is not a valid matter_power_spectrum "
-                                  "type. Available options are: %s" \
-                                 % (matter_power_spectrum, 
-                                    matter_power_spectrum_types.keys()) )
-            if baryons_power_spectrum not in baryons_power_spectrum_types.keys():
-                raise KeyError( "'%s' is not a valid baryons_power_spectrum "
-                                  "type. Available options are: %s" \
-                                 % (baryons_power_spectrum, 
-                                    baryons_power_spectrum_types.keys()) )
+                raise ValueError(
```

Comment:
Why was this made a ValueError instead of a KeyError? Same a few lines below

Reply:
See mike's CR. ValueError is more pythonic. KeyErrors are reserved for dicts.